Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.cpp

```diff
@@ -6,154 +6,119 @@
  *
  */
 #include "FileWatcher.h"
+#include "AzCore/std/containers/vector.h"
 #include <native/assetprocessor.h>
+#include <native/FileWatcher/FileWatcher_platform.h>
+#include <QFileInfo>
 
-//////////////////////////////////////////////////////////////////////////////
-/// FolderWatchRoot
-void FolderRootWatch::ProcessNewFileEvent(const QString& file)
+//! IsSubfolder(folderA, folderB)
+//! returns whether folderA is a subfolder of folderB
+//! assumptions: absolute paths
+static bool IsSubfolder(const QString& folderA, const QString& folderB)
```

Comment:
We should update this comment that this is looking for a sub-directory in the strict sense, that a directory is not a sub-directory of itself.
i.e
`/O3DE/foo` is not a sub-directory of `/O3DE/foo`